mobile/src/viewModels/QoLViewModel.ts

```diff
@@ -130,13 +144,22 @@ export default class QOLSurveyViewModel {
         return res;
     }
 
-    public updateQolOnboarding = () => {
-        this._settings.updateQolOnboarding({ seenQolOnboarding: true, lastFullQol: Date() })
+    public updateQolOnboarding() {
+        this._settings.updateQolSettings({ seenQolOnboarding: true }, 'seenQolOnboarding');
+        this._settings.updateQolSettings({ lastFullQol: Date() }, 'lastFullQol');
         this.showInterlude = true;
     }
 
-    public updatePendingFullQol = () => {
-        this._settings.updatePendingFullQol({ pendingFullQol: false });
+    public updatePendingQol() {
+        switch (this.qolSurveyType) {
```

Comment:
why do we use a switch here as opposed to the ternary used on line 74?

Reply:
When I first wrote this I kept in mind that there might be more types other than Full and Short qol surveys. I will update the other lines to match.